amundsen_application/static/js/ducks/bookmarks/api/v0.ts

```diff
@@ -0,0 +1,39 @@
+import axios, { AxiosResponse, AxiosError } from 'axios';
+
+
+// Add or remove bookmarks for the logged in user
+export function addBookmark(action) {
+  const { key } = action;
+  return axios.put(`/api/metadata/v0/bookmarks?key=${key}`)
+    .then((response: AxiosResponse) => {
+      return response.data;
+    })
+}
+
+export function removeBookmark(action) {
+  const { key } = action;
+  return axios.delete(`/api/metadata/v0/bookmarks?key=${key}`)
+    .then((response: AxiosResponse) => {
+      return response.data;
+    })
+
+}
+
+// Get logged in user's bookmarks
+export function getBookmarks(action) {
+  return axios.get(`/api/metadata/v0/bookmarks`)
+    .then((response: AxiosResponse) => {
+      return response.data;
+    })
+
+}
+
+
+// Get other user's bookmarks
+export function getBookmarksForUser(action) {
```

Comment:
Why do we need separate methods for the logged in user and another user? A user is a user so don't we only need a method that takes the user id and returns bookmarks?

Reply:
Original thought process was in line with the comment above but I think we can fold `getBookmarks` and `getBookmarksForUser` together. We'd still want to keep separate reducer/saga actions since one will update the `myBookmarks` cache.